salt/modules/puppet.py

```diff
@@ -312,7 +313,9 @@ def summary():
 
     try:
         with salt.utils.files.fopen(puppet.lastrunfile, 'r') as fp_:
-            report = salt.utils.yaml.safe_load(fp_)
+            report = salt.utils.yaml.safe_load(
+                salt.utils.stringutils.to_unicode(fp_)
```

Comment:
PyYAML doesn't need the input to be a unicode string, it's actually designed to read from file streams which are not pre-decoded. So this change can be reverted.